### ISSUE TITLE
`code --install-extension` seems to install incompatible extension version

### ISSUE THREAD

**Issue by rjl9** [microsoft/vscode-remote-release#4918]
Downloading and installing the vsix from the marketplace -- VSCode 1.55.2 returns the error "Unable to install extension 'ms-vscode-remote.remote-wsl' as it is not compatible with VS Code '1.55.2'". I am utilizing the VSCode cli command: code --install-extension on a non-internet connected Windows 20h2 enterprise workstation.

**Comment by aeschli** [comment]
Yes, 0.56.0 requires 1.56. 

It looks like `code --install-extension ms-vscode-remote.remote-wsl` installs the the latest version without checking if it is compatible with the current build

**Comment by aeschli** [comment]
Oh, maybe it's only the message that is printed by the console that is misleading:
```
C:\Users\martinae>code --install-extension ms-vscode-remote.remote-wsl
Installing extensions...
Installing extension 'ms-vscode-remote.remote-wsl' v0.56.0...
```

When I open VSCode and check the extensions viewlet I see that the right extension version (0.54.6) is installed.

**Comment by aeschli** [comment]
@rjl9 You trie dto install the 0.56.0 VSIX, and you got the correct error message

**Comment by rjl9** [comment]
@aeschli this is the problem -- we downloaded the latest current version from the marketplace which was 0.56.0. The problem lies in the fact that 0.56.0 depends on VSCode 1.56 which is NOT a stable release. For an enterprise network not connected to the internet, we  install the latest STABLE vscode (at the time was 1.55.2) and we install the latest extensions offered in the marketplace to match that. Unfortunately based on what you are saying it appears that 0.56.0 is a preview/insider release, not a stable release, and is not compatible with the latest production/stable release of VSCode. Why are insider releases pushed as the latest to the marketplace? This seems like they should have a separate extension for preview builds so it does not interfere with stable builds.

**Comment by sandy081** [comment]
It seems user is installing the VSIX which is not compatible. So assigning back to @aeschli for user's question above.

**Comment by aeschli** [comment]
0.56.0 is not just for insiders, it's also for the upcoming 1.56 stable release. 
The problem is that 'Download Extension' button in the marketplace web UI only allows you to download the 'latest' version without telling you what it is for, not giving you the possibility to download other versions.
Ideally there would be a page with all available versions including the information what VSCode version it is for.

That said, the intended way to install extension from the marketplace to use VSCode itsleft (cli, etxnsions viewlet), which knows how to get the proper version.

@sandy081 How do we collect issues against the marketplace?

Closing this issue as works as intended.
@sandy081 Should I create an issue for the misleading cli message?

**Comment by rjl9** [comment]
@aeschli  I agree with the following statement `The problem is that 'Download Extension' button in the marketplace web UI only allows you to download the 'latest' version without telling you what it is for, not giving you the possibility to download other versions.
Ideally there would be a page with all available versions including the information what VSCode version it is for`

I understand what you are saying that this is for the upcoming 1.56 release not just insiders, but why is this released before 1.56 is live when it breaks 1.55.2? 

I disagree with your statement `That said, the intended way to install extension from the marketplace to use VSCode itsleft (cli, etxnsions viewlet), which knows how to get the proper version.` we are on a network without access to internet so installing using vscode does not work. We use the provided vscode cli in the vscode distribution to install the vsix downloaded from the official source. This to is an intended method of installation as not everyone has internet access on enterprise closed networks. The problem does not exist for every package in the marketplace, only the ones that push out builds that are for releases that are not released yet.

**Comment by aeschli** [comment]
> I understand what you are saying that this is for the upcoming 1.56 release not just insiders, but why is this released before 1.56 is live when it breaks 1.55.2?

This allows us to test the new version of the extension already on insiders and have the extension instantaneously ready when a stable build is released. There's no need to synchronize the release of an extension with a VSCode release.
Other extension went the way of having two different extensions, one for insider and stable. That creates a lot of headaches, as there are now two different extension identifiers. Both extension can be installed at the same time and if other extensions depend on you they also need to split into insiders and stable version.

Note, the marketplace already gives access to older versions. What's missing is the information for which VSCode release they are. 

In the case of Remote-WSL, you can tell from the Remote-WSL version number.

- For 1.54, use 0.54.x

![image](https://user-images.githubusercontent.com/6461412/115920531-9c4a1c00-a47a-11eb-82ae-3c3dca235c07.png)

Also, only the last 5 extensions might not be enough if the extension gets many updates.

**Comment by sandy081** [comment]
You can file issues for marketplace here - https://github.com/microsoft/vsmarketplace

> Should I create an issue for the misleading cli message?
Please
